internal/controllers/machine/drain/drain_test.go

```diff
@@ -481,13 +603,124 @@ func TestGetPodsForEviction(t *testing.T) {
 				{
 					Pod: &corev1.Pod{
 						ObjectMeta: metav1.ObjectMeta{
-							Name: "pod-1-delete-multiple-warnings",
+							Name:      "pod-1-delete-multiple-warnings",
+							Namespace: metav1.NamespaceDefault,
 						},
 					},
 					Status: PodDeleteStatus{
-						Delete:  true,
+						DrainBehavior: clusterv1.MachineDrainRuleDrainBehaviorDrain,
+						DrainOrder:    ptr.To[int32](0),
+						Reason:        PodDeleteStatusTypeWarning,
+						Message:       daemonSetOrphanedWarning + ", " + localStorageWarning,
+					},
+				},
+			}},
+		},
+		{
+			name: "drainLabelFilter",
+			pods: []*corev1.Pod{
+				{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      "pod-1-skip-pod-with-drain-label",
+						Namespace: metav1.NamespaceDefault,
+						Labels: map[string]string{
+							clusterv1.PodDrainLabel: "skip",
+						},
+					},
+				},
+			},
+			wantPodDeleteList: PodDeleteList{items: []PodDelete{
+				{
+					Pod: &corev1.Pod{
+						ObjectMeta: metav1.ObjectMeta{
+							Name:      "pod-1-skip-pod-with-drain-label",
+							Namespace: metav1.NamespaceDefault,
+						},
+					},
+					Status: PodDeleteStatus{
+						DrainBehavior: clusterv1.MachineDrainRuleDrainBehaviorSkip,
+						Reason:        PodDeleteStatusTypeSkip,
+					},
+				},
+			}},
+		},
+		{
+			name: "machineDrainRulesFilter - error: Namespace does not exist",
+			pods: []*corev1.Pod{
+				{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      "pod-1-non-existing-namespace",
+						Namespace: "non-existing-namespace",
+					},
+				},
+			},
+			wantErr: "failed to get Pods for eviction: Pods with error \"Pod Namespace does not exist\": non-existing-namespace/pod-1-non-existing-namespace",
```

Comment:
wondering about this use case, this seems more an internal error (e.g. cache out of sync) than an error we want to surface to users 🤔

Reply:
Maybe race conditions are possible not sure. It should almost never happen, agree. But I think I also can't really ignore this error case